core/core.mk

```diff
@@ -40,6 +40,14 @@ cflags_kasan	+= -fsanitize=kernel-address \
 		   --param asan-instrumentation-with-call-threshold=0
 cflags$(sm)	+= $(cflags_kasan)
 endif
+ifeq ($(CFG_CORE_CHECK_STACKS),y)
+finstrument-functions := $(call cc-option,-finstrument-functions)
+ifeq (,$(finstrument-functions))
+$(warning -finstrument-functions not supported, disabling CFG_CORE_CHECK_STACKS)
+override CFG_CORE_CHECK_STACKS := n
```

Comment:
I think this should rather fail the build and let caller decide if `CFG_CORE_CHECK_STACKS` should be disabled or toolchain changed.

Reply:
Yes. I did that because I initially wanted to enable it with `DEBUG=1` and I did not want to break existing builds depending on the compiler used. But in the end I chose not to do that (it's not very good if DEBUG means different things based on the compiler, same argument as yours).
Let's simplify this and error out instead.